src/test/java/utils/GeneralConfigModule.java

```diff
@@ -21,15 +21,28 @@ public void applyToConfigBuilder(SchemaGeneratorConfigBuilder builder) {
 		builder.forTypesInGeneral()
 				.withSubtypeResolver(new ClassGraphSubtypeResolver())
 				.withCustomDefinitionProvider(descriptionProvider)
-				.withTypeAttributeOverride(descriptionProvider);
+				.withTypeAttributeOverride(descriptionProvider)
+				//.withCustomDefinitionProvider(new ObjectIdToCustomSchemaProvider())
+		;
+		builder.forFields()
+				.withTargetTypeOverridesResolver(this::collectTargetTypeOverrides);
 		builder.forMethods()
+				.withTargetTypeOverridesResolver(this::collectTargetTypeOverrides)
 				.withIgnoreCheck(method -> method.getSchemaPropertyName().endsWith(")"));
-		builder
-//				.with(new SimpleTypeModule().withStringType(ObjectId.class))
-				.forTypesInGeneral()
-				.withCustomDefinitionProvider(new ObjectIdToCustomSchemaProvider())
-		;
+	}
 
+	private List<ResolvedType> collectTargetTypeOverrides(MemberScope<?, ?> member) {
+		if (member.getType() == null || !member.getType().isInstanceOf(ObjectId.class)) {
+			return null;
+		}
+		return Stream.of(String.class, ObjectIdRepresentation.class)
+				.map(member.getContext()::resolve)
+				.collect(Collectors.toList());
+	}
+
+	private static class ObjectIdRepresentation {
+		@NotNull protected int date;
+		@NotNull protected int timestamp;
```

Comment:
`@NotNull` annotations and inclusion of `JakartaValidationModule` with corresponding validation API are meant to mark those fields as required.
If this is only necessary here, a simpler `withRequiredCheck()` could be configured instead.